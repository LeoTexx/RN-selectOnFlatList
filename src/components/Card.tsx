import React from 'react';
import {Text, TouchableOpacity, StyleSheet} from 'react-native';

interface CardProps {
  selected: boolean;
  onPress: Function;
  item: {
    id: number;
    firstName: string;
    lastName: string;
    role: string;
    description: string;
    photo: string;
  };
}

export default function Card({item, onPress, select}): CardProps {
  return (
    <TouchableOpacity
      testID="Card"
      onPress={onPress}
      style={
        item.id === select
          ? [styles.container, {backgroundColor: 'rgba(173, 216, 230,0.8)'}]
          : [styles.container]
      }
      key={item.id}>
      <Text style={styles.title}>
        <Text style={{fontWeight: 'bold'}}>{item.firstName}</Text>{' '}
        {item.lastName}
      </Text>
      <Text style={styles.role}>{item.role}</Text>
      <Text style={styles.description}>
        <Text style={{fontWeight: 'bold'}}>About me{'\n'}</Text>
        {item.description}
      </Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 32,
    borderWidth: 5,
    borderColor: '#add8e6',
    borderRadius: 48,
    height: 512,
    marginVertical: 36,
  },
  title: {
    alignItems: 'center',
    justifyContent: 'center',
    fontSize: 32,
  },
  role: {
    fontSize: 24,
  },
  description: {},
});
