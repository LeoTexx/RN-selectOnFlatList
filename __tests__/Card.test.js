import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import Card from '../src/components/Card';

const view = renderer.create(
  <Card select={true} onPress={jest.fn()} item={jest.fn()} />,
);

it('Test the rendering of the Card component', () => {
  expect(view.toJSON()).toMatchSnapshot();
});
