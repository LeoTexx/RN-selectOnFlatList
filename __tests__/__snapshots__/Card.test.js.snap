// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the rendering of the Card component 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#add8e6",
      "borderRadius": 48,
      "borderWidth": 5,
      "flex": 1,
      "height": 512,
      "justifyContent": "center",
      "marginVertical": 36,
      "opacity": 1,
      "padding": 32,
    }
  }
  testID="Card"
>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 32,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
     
  </Text>
  <Text
    style={
      Object {
        "fontSize": 24,
      }
    }
  />
  <Text
    style={Object {}}
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      About me
      

    </Text>
  </Text>
</View>
`;
